utils/components-utils/src/main/scala/pl/touk/nussknacker/engine/util/sinkvalue/SinkValueData.scala

```diff
@@ -31,13 +38,35 @@ object SinkValueData {
 
   sealed trait SinkValueParameter {
     def toParameters: List[Parameter] = this match {
-      case SinkSingleValueParameter(value) => value :: Nil
+      case SinkSingleValueParameter(value, _) => value :: Nil
       case SinkRecordParameter(fields) => fields.values.toList.flatMap(_.toParameters)
     }
+    def validateParams(resultType: List[(String, BaseDefinedParameter)], prefix: List[String])(implicit nodeId: NodeId): ValidatedNel[ProcessCompilationError, Unit]
   }
 
-  case class SinkSingleValueParameter(value: Parameter) extends SinkValueParameter
+  case class SinkSingleValueParameter(value: Parameter, validator: SinkValueValidator) extends SinkValueParameter {
+    override def validateParams(resultTypes: List[(String, BaseDefinedParameter)], prefix: List[String])(implicit nodeId: NodeId): ValidatedNel[ProcessCompilationError, Unit] = resultTypes match {
```

Comment:
I have one confusion about how this `validateParams` is related to `Parameter.typ`. Am I right that we do two validations:
- against this `Parameter.typ` in our "engine"
- using `validateParams` in component?
If so, IMO we should write some comment about this relation eg. that this `Parameter.typ` should be "wider" than  `validateParams` to not limit this validations.